compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3189,13 +3199,16 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
         matchEnd: LabelDef)(implicit pos: Position): js.Tree = {
 
       val matchEndSym = matchEnd.symbol
-      countsOfReturnsToMatchEnd(matchEndSym) = 0
+      countsOfReturnsToPatmatLabel(matchEndSym) = 0
 
       val nextCaseSyms = (cases.tail map (_.symbol)) :+ NoSymbol
 
       val translatedCases = for {
         (LabelDef(_, Nil, rhs), nextCaseSym) <- cases zip nextCaseSyms
       } yield {
+        if (nextCaseSym.exists)
+          countsOfReturnsToPatmatLabel(nextCaseSym) = 0
```

Comment:
Is there a reason we are not using a different map for this? I think it would aid understanding quite a bit.